wallet/src/main/scala/org/bitcoins/wallet/models/DLCDAO.scala

```diff
@@ -0,0 +1,47 @@
+package org.bitcoins.wallet.models
+
+import org.bitcoins.core.crypto.Sha256DigestBE
+import org.bitcoins.db.{CRUD, SlickUtil}
+import org.bitcoins.wallet.config._
+import slick.jdbc.SQLiteProfile.api._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+case class DLCDAO()(
+    implicit val ec: ExecutionContext,
+    val appConfig: WalletAppConfig)
+    extends CRUD[DLCDb, Sha256DigestBE] {
+  import org.bitcoins.db.DbCommonsColumnMappers._
+
+  override val table: TableQuery[DLCTable] = TableQuery[DLCTable]
+
+  override def createAll(ts: Vector[DLCDb]): Future[Vector[DLCDb]] =
+    SlickUtil.createAllNoAutoInc(ts, database, table)
+
+  override protected def findByPrimaryKeys(
+      ids: Vector[Sha256DigestBE]): Query[Table[_], DLCDb, Seq] =
+    table.filter(_.eventId.inSet(ids))
+
+  override def findByPrimaryKey(
+      id: Sha256DigestBE): Query[Table[_], DLCDb, Seq] = {
+    table
+      .filter(_.eventId === id)
+  }
+
+  override def findAll(dlcs: Vector[DLCDb]): Query[Table[_], DLCDb, Seq] =
+    findByPrimaryKeys(dlcs.map(_.eventId))
+
+  def findByEventId(eventId: Sha256DigestBE): Future[Option[DLCDb]] = {
```

Comment:
I know I pushed back against this in our meeting yesterday, but if we do decide to go with the `eventId` as the primary key, I believe this method is the same thing as using `CRUD.read()` ?